simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -850,5 +850,151 @@ auto EntityBase::updateTraveledDistance(const double step_time) -> double
+    static_cast<LaneletPose>(target_lanelet_pose_),
+    static_cast<LaneletPose>(entity_lanelet_pose_.value()));
+
+  if (!revert_entity_distance_to_target.has_value() && !entity_distance_to_target.has_value()) {
+    THROW_SIMULATION_ERROR("Failed to get distance between entity and target lanelet pose.");
+  }
+
+  if (!revert_entity_distance_to_target.has_value()) {
+    return entity_distance_to_target.value();
+  }
+
+  if (!entity_distance_to_target.has_value()) {
+    return revert_entity_distance_to_target.value();
+  }
+
+  if (revert_entity_distance_to_target.value() < entity_distance_to_target.value()) {
+    return revert_entity_distance_to_target.value();
+  } else {
+    return entity_distance_to_target.value();
+  }
+}
+
+auto EntityBase::isArrivedToTargetLaneletPose(
+  const CanonicalizedLaneletPose & target_lanelet_pose, const double threshold) -> bool
+{
+  const auto entity_distance_to_target = getDistanceToTargetLaneletPose(target_lanelet_pose);
+
+  if (entity_distance_to_target <= threshold) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+auto EntityBase::requestSynchronize(
+  const CanonicalizedLaneletPose & ego_target, const CanonicalizedLaneletPose & entity_target,
+  const double threshold, const double accel_limit, const double loop_period) -> bool
+{
+  if (traffic_simulator_msgs::msg::EntityType::EGO == getEntityType().type) {
+    THROW_SYNTAX_ERROR("Request synchronize is only for non-ego entities.");
+  }
+
+  /**
+   * @brief Check if the entity has already arrived to the target lanelet.
+   */
+  if (isArrivedToTargetLaneletPose(entity_target, threshold)) {
+    if (this->getStatus().getTwist().linear.x < accel_limit * loop_period / 1000) {
+    } else {
+      RCLCPP_WARN_ONCE(
+        rclcpp::get_logger("traffic_simulator"),
+        "The entity has already arrived to the target lanelet and the entity is not nearly "
+        "stopped.");
+    }
+    target_speed_ = 0;
+    return true;
+  }
+
+  job_list_.append(
+    [this, ego_target, entity_target, accel_limit, threshold, loop_period](double) {
+      /**
+       * @brief This is function for synchronization.
+       */
+      auto entity_distance = getDistanceToTargetLaneletPose(entity_target);
+      auto ego_distance = this->hdmap_utils_ptr_->getLongitudinalDistance(
+        static_cast<LaneletPose>(other_status_.find("ego")->second.getLaneletPose()),
+        static_cast<LaneletPose>(ego_target));
+
+      if (!ego_distance.has_value()) {
+        RCLCPP_WARN_ONCE(
+          rclcpp::get_logger("traffic_simulator"),
+          "ego has already passed away or has some problem");
+        return true;
+      }
+      const auto ego_velocity = other_status_.find("ego")->second.getTwist().linear.x;
+      const auto entity_velocity = this->getStatus().getTwist().linear.x;
+      const auto ego_arrival_time = (ego_velocity != 0) ? ego_distance.value() / ego_velocity : 0;
```

Comment:
To determine if it is equal to 0 for a double type, [consider the machine epsilon.](https://cpprefjp.github.io/reference/limits/numeric_limits/epsilon.html)